src/EXTRA-FIX/fix_wall_flow.cpp

```diff
@@ -0,0 +1,328 @@
+  if (rndseed <= 0)
+    error->all(FLERR, "Illegal fix wall/flow argument: random seed must be positive integer");
+
+  ++iarg;
+  // parsing wall count
+  int wallcount = utils::inumeric(FLERR, arg[iarg], do_abort, lmp);
+  if (wallcount <= 0)
+    error->all(FLERR, "Illegal fix wall/flow argument: wall count must be positive integer");
+
+  ++iarg;
+  // parsing walls
+  if (narg - iarg != wallcount && narg - iarg != wallcount + 2)
+    error->all(FLERR, "Wrong fix wall/flow wall count");
+
+  double scale = 0.0;
+  if (flowax == FlowAxis::AX_X)
+    scale = domain->lattice->xlattice;
+  else if (flowax == FlowAxis::AX_Y)
+    scale = domain->lattice->ylattice;
+  else if (flowax == FlowAxis::AX_Z)
+    scale = domain->lattice->zlattice;
+
+  if (narg - iarg == wallcount + 2) {
+    if (strcmp(arg[narg - 2], "units") != 0) error->all(FLERR, "Wrong fix wall/flow units command");
+    if (strcmp(arg[narg - 1], "box") == 0)
+      scale = 1.0;
+    else if (strcmp(arg[narg - 1], "lattice") != 0)
+      error->all(FLERR, "Wrong fix wall/flow units command");
+  }
+
+  walls.resize(wallcount + 2);
+  walls.front() = domain->boxlo[flowax];
+  for (int w = 1; w <= wallcount; ++w, ++iarg) {
+    walls[w] = utils::numeric(FLERR, arg[iarg], do_abort, lmp) * scale;
+  }
+  walls.back() = domain->boxhi[flowax];
+  if (!std::is_sorted(walls.begin(), walls.end(), std::less_equal<double>())) {
+    error->all(FLERR,
+               "Wrong fix wall/flow wall ordering or some walls are outside simulation domain");
+  }
+
+  if (std::adjacent_find(walls.begin(), walls.end()) != walls.end()) {
+    error->all(FLERR,
+               "Wrong fix wall/flow wall coordinates: some walls have the same coordinates or lie "
+               "on the boundary");
+  }
+
+  memory->grow(current_segment, atom->nmax, "WallFlow::current_segment");
+  atom->add_callback(Atom::GROW);
+  if (restart_peratom) atom->add_callback(Atom::RESTART);
+
+  maxexchange = 1;
+
+  random = new RanMars(lmp, rndseed + comm->me);
+}
+
+/* ---------------------------------------------------------------------- */
+
+FixWallFlow::~FixWallFlow()
+{
+  if (copymode) return;
+  atom->delete_callback(id, Atom::GROW);
+  if (restart_peratom) atom->delete_callback(id, Atom::RESTART);
+  memory->destroy(current_segment);
+
+  delete random;
+}
+
+/* ---------------------------------------------------------------------- */
+
+int FixWallFlow::setmask()
+{
+  int mask = 0;
+
+  mask |= END_OF_STEP;
+
+  return mask;
+}
+
+/* ---------------------------------------------------------------------- */
+
+void FixWallFlow::init()
+{
+  int nrigid = 0;
+  int box_change_flowax = 0;
+  for (const auto &ifix : modify->get_fix_list()) {
+    if (ifix->rigid_flag) nrigid++;
+    switch (flowax) {
+      case FlowAxis::AX_X:
+        if (ifix->box_change & Fix::BOX_CHANGE_X) box_change_flowax++;
+        if (ifix->box_change & Fix::BOX_CHANGE_XY) box_change_flowax++;
```

Comment:
Since the box cannot be triclinic, there is no need to check in init() for XY, ZX, YZ